src/MultiIndex.zig

```diff
@@ -59,7 +61,17 @@ pub const IndexRef = struct {
     references: usize = 1,
     delete_files: bool = false,
     being_deleted: bool = false,
+    load_state: LoadState = .ready,
+    load_context: ?LoadContext = null,
```

Comment:
I don't think we even need this, we can just pass generation and snapshot_url to the task as additional parameters.